### PR TITLE
ECER-1513 - Character/work ex ref submission - mapped date signed pro…

### PR DESCRIPTION
**Pull request by qsl-deepak**
…p, added null/empty checks for reference selected province id

---
name: ECER-1513 - Connect FE to BE
about: Character/work ex ref submission - mapped date signed prop, added null/empty checks for reference selected province id
---

## Title
ECER-1513: Connect FE to BE

## Description

- mapped "Date Signed" prop through all layers from end point to dynamics
- added null/empty checks incases where reference selected province id is null/empty

## Related Jira Issue(s)

- ECER-1523
- ECER-873

## Checklist
- [x] I have tested these changes locally.
- [x] Changes to backend endpoints are covered by passing integration tests.
- [ ] I have added or updated the necessary documentation.

### CODE REVIEW

File: src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/References/ReferencesEndpoints.cs [hunk 1]
```diff
@@ -53,11 +53,13 @@ public void Register(IEndpointRouteBuilder endpointRouteBuilder)
   }
 }
 
-public record CharacterReferenceSubmissionRequest(string Token, bool WillProvideReference, ReferenceContactInformation ReferenceContactInformation, CharacterReferenceEvaluation ReferenceEvaluation, bool ConfirmProvidedInformationIsRight);
-public record ReferenceContactInformation([Required] string LastName, [Required] string FirstName, [Required] string Email, [Required] string PhoneNumber, string CertificateProvinceId, string CertificateProvinceOther)
+public record CharacterReferenceSubmissionRequest(string Token, bool WillProvideReference, ReferenceContactInformation ReferenceContactInformation, CharacterReferenceEvaluation ReferenceEvaluation, bool ConfirmProvidedInformationIsRight, DateTime DateSigned);
```
Comment (ytqsl) [inline src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/References/ReferencesEndpoints.cs:56]: The submission date is not something you'd want the UI to set, it is a logical value set by at manager at the time of submission

File: src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/References/ReferencesEndpoints.cs [hunk 2]
```diff
@@ -73,7 +75,7 @@ public enum UnabletoProvideReferenceReasons
...
-public record WorkExperienceReferenceSubmissionRequest([Required] string Token, bool WillProvideReference, ReferenceContactInformation ReferenceContactInformation, WorkExperienceReferenceDetails WorkExperienceReferenceDetails, WorkExperienceReferenceCompetenciesAssessment WorkExperienceReferenceCompetenciesAssessment, [Required] bool ApplicantShouldNotBeECE, string ApplicantNotQualifiedReason, bool ConfirmProvidedInformationIsRight);
+public record WorkExperienceReferenceSubmissionRequest([Required] string Token, bool WillProvideReference, ReferenceContactInformation ReferenceContactInformation, WorkExperienceReferenceDetails WorkExperienceReferenceDetails, WorkExperienceReferenceCompetenciesAssessment WorkExperienceReferenceCompetenciesAssessment, [Required] bool ApplicantShouldNotBeECE, string ApplicantNotQualifiedReason, bool ConfirmProvidedInformationIsRight, DateTime DateSigned);
```
Comment (ytqsl) [inline src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/References/ReferencesEndpoints.cs:78]: Same as above

File: src/ECER.Managers.Registry.Contract/Applications/Contract.cs [hunk 3]
```diff
@@ -119,9 +119,10 @@ public enum ApplicationStatus
   ReconsiderationDecision
 }
 
-public record CharacterReferenceSubmissionRequest(string Token, bool WillProvideReference, ReferenceContactInformation ReferenceContactInformation, CharacterReferenceEvaluation ReferenceEvaluation, bool ConfirmProvidedInformationIsRight);
-public record ReferenceContactInformation(string LastName, string FirstName, string Email, string PhoneNumber, string CertificateProvinceId, string CertificateProvinceOther)
+public record CharacterReferenceSubmissionRequest(string Token, bool WillProvideReference, ReferenceContactInformation ReferenceContactInformation, CharacterReferenceEvaluation ReferenceEvaluation, bool ConfirmProvidedInformationIsRight, DateTime DateSigned);
```
Comment (ytqsl) [inline src/ECER.Managers.Registry.Contract/Applications/Contract.cs:122]: same

File: src/ECER.Resources.Documents/Applications/ApplicationRepository.cs [hunk 4]
```diff
@@ -237,10 +237,13 @@ private async Task<string> SubmitCharacterReference(string characterReferenceId,
     var characterReference = context.ecer_CharacterReferenceSet.Single(c => c.ecer_CharacterReferenceId == Guid.Parse(characterReferenceId));
 
     mapper.Map(request, characterReference);
-    var province = context.ecer_ProvinceSet.SingleOrDefault(p => p.ecer_ProvinceId == Guid.Parse(request.ReferenceContactInformation.CertificateProvinceId));
-    if (province != null)
+    if (!string.IsNullOrWhiteSpace(request.ReferenceContactInformation.CertificateProvinceId))
     {
-      context.AddLink(characterReference, ecer_CharacterReference.Fields.ecer_characterreference_RefCertifiedProvinceId, province);
+      var province = context.ecer_ProvinceSet.SingleOrDefault(p => p.ecer_ProvinceId == Guid.Parse(request.ReferenceContactInformation.CertificateProvinceId));
```
Comment (ytqsl) [inline src/ECER.Resources.Documents/Applications/ApplicationRepository.cs:242]: What happens when this is not a guid? is province mandatory?

Comment (qsl-deepak) [reply src/ECER.Resources.Documents/Applications/ApplicationRepository.cs]: Province is optional, so Province Id can possibly have following value types -> empty, null or a GUID.
Added Guid try parse check.

**Review by ytqsl — APPROVED** [review]

**Pull request merged**